### PR TITLE
Minor fixes for `ConvertTorchConversionToMLProgram`.

### PR DESCRIPTION
**Pull request by arames**
* Only create the global seed variable if it does not exist already.
* Make the pass a module pass. A func pass may not modify its parent op.

### CODE REVIEW

**Comment by arames** [comment]
The current implementation works when there is  only a single function in the input module.

File: test/Conversion/TorchConversionToMLProgram/multiple_functions.mlir [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// RUN: torch-mlir-opt %s -convert-torch-conversion-to-mlprogram -split-input-file | FileCheck %s
+
+module {
+  func.func private @f0() -> i64
```
Comment (arames) [inline test/Conversion/TorchConversionToMLProgram/multiple_functions.mlir:4]: A higher number of functions helps trigger the race when multiple threads process them concurrently.

**Review by vivekkhandelwal1 — APPROVED** [review]
LGTM!

**Comment by silvasean** [comment]
Good catch! Thanks!

**Pull request merged**